microprot/scripts/pdb_search.py

```diff
@@ -0,0 +1,393 @@
+_HEADER = ['No',
+           'Hit',
+           'Prob',
+           'E-value',
+           'P-value',
+           'Score',
+           'SS',
+           'Cols',
+           'Query HMM',
+           'Template HMM',
+           '#match states']
+
+
+def is_overlapping(intA, intB):
+    ''' Test if two intervals overlap.
+
+    Parameters
+    ----------
+    intA : (int, int)
+        A pair of coordinates, e.g. (20, 100). Left component must be <= right.
+    intB : (int, int)
+        A pair of coordinates, e.g. (50, 150). Left component must be <= right.
+
+    Returns
+    -------
+    Bool. True if intA overlaps with intB. False otherwise.
+
+    Raises
+    ------
+    ValueError
+        If left component of intA or intB is greater than its right component.
+    '''
+    if (intA[0] > intA[1]):
```

Comment:
To avoid duplicated code, how about:
```
for i, x in enumerate((intA, intB)):
    if (x[0] > x[1]):
        raise ValueError("Left component of integer %s cannot be larger than "
                         "its right component, i.e. the interval must be "
                         "forward oriented." % chr(i+65))
```